swift_scality_backend/diskfile.py

```diff
@@ -192,6 +193,11 @@ def zero_copy_send(self, wsockfd):
 
             try:
                 conn.putrequest('GET', object_url.path, skip_host=False)
+                if client._url_username and client._url_password:
+                    creds_str = ('%s:%s' % (client._url_username, client._url_password))
+                    basic_auth_header = get_urllib3().util.make_headers(basic_auth=creds_str)
+                    conn.putheader(basic_auth_header.keys()[0],
```

Comment:
Ugh! Please consider
```suggestion
basic_auth_headers = get_urllib3().util.make_headers(basic_auth=creds_str)
for (key, value) in basic_auth_headers.items():
    conn.putheader(key, value)
```
instead of relying on 'knowing' what the headers look like...

Reply:
Fixed